File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0)
            return false;
            
        int rev = 0;
        int temp = x;

        while (temp != 0) {
            int lastDigit = temp % 10;
            rev = rev * 10 + lastDigit;
            temp = temp / 10;
        }
        return rev == x;
    }
}

File: 9-palindrome-number/palindrome-number.ts
function isPalindrome(x: number): boolean {
    const num = x;
    if (x < 0) return false;

    let ans = 0;

    while (x > 0) {
        let lastDigit = x % 10;
        ans = 10 * ans + lastDigit;
        x = Math.floor(x / 10);
    }
    return num === ans;
};